src/Sentry/TransactionTracer.cs

```diff
@@ -178,21 +180,35 @@ public IReadOnlyList<string> Fingerprint
     /// Initializes an instance of <see cref="Transaction"/>.
     /// </summary>
     public TransactionTracer(IHub hub, string name, string operation)
-        : this(hub, name, operation, TransactionNameSource.Custom)
+        : this(hub, name, operation, TransactionNameSource.Custom, null)
     {
     }
 
     /// <summary>
     /// Initializes an instance of <see cref="Transaction"/>.
     /// </summary>
-    public TransactionTracer(IHub hub, string name, string operation, TransactionNameSource nameSource)
+    public TransactionTracer(IHub hub, string name, string operation, TransactionNameSource nameSource, TimeSpan? idleTimeout)
```

Comment:
@bitsandfoxes, do you know why there's a 3rd constructor that is not chained? 
https://github.com/getsentry/sentry-dotnet/blob/1074-transacton-idle-timeout/src/Sentry/TransactionTracer.cs#L217

Reply:
`TransactionContext` contains a superset of the properties being passed in the other constructors... maybe everything should ultimately chain up to that? It's the one that gets used [in the Hub](https://github.com/getsentry/sentry-dotnet/blob/5401fed146c2219f2eae25cf26b96419e59b804e/src/Sentry/Internal/Hub.cs#L119).

Reply:
I got confused by the 3 constructors
1.  `public TransactionTracer(IHub hub, string name, string operation)` - calling the 2nd constructor
2. `public TransactionTracer(IHub hub, string name, string operation, TransactionNameSource nameSource, TimeSpan? idleTimeout)`
3. `public TransactionTracer(IHub hub, ITransactionContext context)`

The [1st constructor](https://github.com/getsentry/sentry-dotnet/blob/7db3f1b9e8569986d3929e27310f41140fa19723/src/Sentry/TransactionTracer.cs#L180) is chaining to the [2nd constructor](https://github.com/getsentry/sentry-dotnet/blob/7db3f1b9e8569986d3929e27310f41140fa19723/src/Sentry/TransactionTracer.cs#L188).
The [3rd constructor](https://github.com/getsentry/sentry-dotnet/blob/7db3f1b9e8569986d3929e27310f41140fa19723/src/Sentry/TransactionTracer.cs#L201), using `ITransactionContext` is not chaining to anything.

The 1st constructor is mainly used for testing purposes. It also solely uses the 2nd constructor.
This supports what you're suggesting, @jamescrosswell. 

Why do we need the 1st (and the 2nd) constructor as a public constructor, given it's for testing only and could be made internal? Potentially, those two could be made internal in the next major.

Reply:
Sounds reasonably. I'm not sure if people using the SDK fiddle with TransactionTracers directly at all tbh - I imagine they start transactions and create child spans from the SDK. The Tracers seem like an implementation detail (it's what we wrap those transactions/spans in so that we've got somewhere to track performance data).

Reply:
I'd like to leave the 1st and the 2nd contractors disconnected from the 3rd one, which the Hub uses anyway. If those two are solely for testing purposes, perhaps some comment could be made to clarify their purpose, along with a TODO for version 4 to make those internal.

Reply:
> The 1st constructor is mainly used for testing purposes. 
> Why do we need the 1st (and the 2nd) constructor as a public constructor, given it's for testing only and could be made internal? Potentially, those two could be made internal in the next major.

Agree! These should be `internal`, we can use them for testing through `InternalsVisibleTo` and they don't polute the public API

Reply:
@bruno-garcia, aside from leaving a to-do comment in the code, is there anything else that would be used to indicate future deprecation?

Reply:
@bruno-garcia, should I raise an issue for v4 to mark those methods as internal? Or is there a registry for v4 changes this could be added to? /cc @bitsandfoxes